### PR TITLE
Add new dataset: GermanGovServiceRetrieval

### PR DESCRIPTION
**Pull request by malteos**
GermanGovServiceRetrieval: LHM-Dienstleistungen-QA is a German question answering dataset for government services of the Munich city administration. It associates questions with a textual context containing the answer

## Checklist for adding MMTEB dataset

Reason for dataset addition: Domain-specific retrieval dataset for German

- [x] I have tested that the dataset runs with the `mteb` package.
- [x] I have run the following models on the task (adding the results to the pr). These can be run using the `mteb run -m {model_name} -t {task_name}` command.
  - [x] `sentence-transformers/paraphrase-multilingual-MiniLM-L12-v2`
  - [x] `intfloat/multilingual-e5-small`
- [x] I have checked that the performance is neither trivial (both models gain close to perfect scores) nor random (both models gain close to random scores).
- [x] If the dataset is too big (e.g. >2048 examples), considering using `self.stratified_subsampling() under dataset_transform()`
- [x] I have filled out the metadata object in the dataset file (find documentation on it [here](https://github.com/embeddings-benchmark/mteb/blob/main/docs/adding_a_dataset.md#2-creating-the-metadata-object)).
- [x] Run tests locally to make sure nothing is broken using `make test`. 
- [x] Run the formatter to format the code using `make lint`. 
- [x] I have added points for my submission to the [points folder](https://github.com/embeddings-benchmark/mteb/blob/main/docs/mmteb/points.md) using the PR number as the filename (e.g. `438.jsonl`).

### CODE REVIEW

**Comment by malteos** [comment]
Do you have time to review? @guenthermi @Muennighoff @PhilipMay

**Review by KennethEnevoldsen — APPROVED** [review]
I think everything looks good. Feel free to add points. You might also consider using ndcg_5 instead of 10 since the dataset is quite small.

**Comment by malteos** [comment]
> I think everything looks good. Feel free to add points. You might also consider using ndcg_5 instead of 10 since the dataset is quite small.

Good point. I changed the main metric.

**Comment by PhilipMay** [comment]
> Do you have time to review? @guenthermi @Muennighoff @PhilipMay

Hey Malte. I have almost no knowledge with MTEB. Never implemented anything here. Sorry.
Maybe @rasdani could have a look?

File: docs/mmteb/points/731.jsonl [hunk 1]
```diff
@@ -0,0 +1,2 @@
+{"GitHub": "malteos", "New dataset": 6}
```
Comment (KennethEnevoldsen) [inline docs/mmteb/points/731.jsonl:1]:
```suggestion
{"GitHub": "malteos", "New dataset": 2}
```
I believe a German retrieval task already exists in XQuADRetrieval

**Review by KennethEnevoldsen — APPROVED** [review]
Have enabled auto-merge and updated the points. Let me know if you disagree. Thanks for the addition!

**Pull request merged**